app/src/lib/stores/api-repositories-store.ts

```diff
@@ -4,6 +4,41 @@ import { IAPIRepository, API } from '../api'
 import { Account } from '../../models/account'
 import { merge } from '../merge'
 
+/**
+ * Returns a value indicating whether two account instances
+ * can be considered equal. Equality is determined by comparing
+ * the two instances' endpoints and user id. This allows
+ * us to keep receiving updated Account details from the API
+ * while still maintaining the association between repositories
+ * and a particular account.
+ */
+function accountEquals(x: Account, y: Account) {
+  return x.endpoint === y.endpoint && x.id === y.id
+}
+
+/**
+ * Attempt to look up an existing account in the account state
+ * map based on endpoint and user id equality (see accountEquals).
+ *
+ * If no match is found the provided account is returned.
+ */
+function resolveAccount(
+  account: Account,
+  accountState: ReadonlyMap<Account, IAccountRepositories>
+) {
+  if (accountState.has(account)) {
+    return account
+  }
+
+  for (const existingAccount of accountState.keys()) {
+    if (accountEquals(existingAccount, account)) {
```

Comment:
A comment here stating why we go from reference equality to value equality would help clarify the need to iterate through accounts after checking the set for it. ✨

Reply:
Pushed an expanded documentation, let me know if that feels better.